Feat: Migrated to React-Navigation v5

IronTony opened the pull request:
Removed React-Navigation v4
Added React-Navigation v5
Rewritten routes file
Added modal page

panz3r commented on src/scenes/ModalPage/index.tsx:
```diff
@@ -0,0 +1,37 @@
+
+const ModalPage = () => {
+  const [t] = useTranslation();
+  const navigation = useContext(NavigationContext);
```
`useContext(NavigationContext)` can be replaced with `useNavigation()`

IronTony commented:
Fixed some navigation calls
Refactorized ModalPage
Changed navigation from right for pages on android
Added new style of transition for the modal page

IronTony commented on src/components/GenericModal/index.tsx:
```diff
@@ -0,0 +1,35 @@
+
+const ModalPage = ({ children, pageTitle }) => {
+  const navigation = useNavigation();
+  const popAction = StackActions.pop();
```
Done but still not happy with that verbose solution, as we already discussed

IronTony commented on src/scenes/OtherPage/index.tsx:
```diff
@@ -1,13 +1,15 @@
+const OtherPage = () => {
   const [t, i18n] = useTranslation();
+  const navigation = useNavigation();
+  const popAction = StackActions.pop();
```
Done but still not happy with that verbose solution, as we already discussed

panz3r commented:
Related to the `back` question a shorter version could be achieved by creating a custom hook, like this
```ts
function useNavigationPopAction(count: number = 1) {
  const navigation = useNavigation();
  const goBack = useCallback(() => {
    const popAction = StackActions.pop(count);
    navigation.dispatch(popAction);
  }, [count, navigation]);
  return goBack;
}
```

And using it like this
```ts
const goBack = useNavigationPopAction();
.
.
<Button
  onClick={goBack}
/>
.
.
```
